torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3955,6 +3955,48 @@ def sample_inputs_matmul(op_info, device, dtype, requires_grad):
     return tuple(sample_inputs)
 
 
+def sample_inputs_meshgrid(
```

Comment:
> What's up with this particular sample inputs structure? I think all the other sample inputs don't have to worry about returning a callable

it is subtle enough to require documentation. I am generating two opinfos: one that calls it with a list of tensors, and one that calls it variadically. I split this up because using numpy as a reference implementation only works with variadic tensors.

Some alternatives:
1) copy and paste the input generation to two separate functions, one to produce sample inputs that are lists and the other that has *args. I have to live with myself though.
2) write it once and transform the sample inputs after generation rather than during. Such an approach might actually reusable for other operators with similar profiles.
3) rather than return a callable, return the sample inputs and wrap that in a lambda when creating the op infos. This is a little cleaner because we don't have to create a function that returns a function. This is probably the cleanest alternative, but it would make the opinfo slightly different than others. But it already does with the calling convention param.
4) only test one of *args vs list[Tensor]. If list[Tensor] is chosen, then write an adaptor over the reference implementation. This was my initial approach, it worked fine, but I think it's worthwhile to test both calling conventions.
5) only support the *args calling convention in meshgrid itself. IMO this one makes sense for ultimate numpy/scipy/python array api compatibility, but is itself BC breaking and is perhaps not PyTorchic.
6) annotate the return type with a protocol, rather than a callable.

Of these options, I prefer 5 in the long-term, especially if we're chaining some BC breaking changes to meshgrid already. I can't speak to how PyTorch appropriate that is though.

3) and 6) I think are both fairly clean. 3 is probably a tad cleaner though since it should avoid the nested function.

What do you think?

Reply:
OK, per-offline conversation, I used partial to achieve this. Very clean, thanks!